tests/pytorch/test_pytorch_autolog.py

```diff
@@ -262,15 +262,17 @@ def test_pytorch_autolog_non_early_stop_callback_does_not_log(pytorch_model):
 
 @pytest.fixture
 def pytorch_model_tests():
+    mlflow.pytorch.autolog()
     model = IrisClassification()
     dm = IrisDataModule()
     dm.prepare_data()
     dm.setup(stage="fit")
     trainer = pl.Trainer(max_epochs=NUM_EPOCHS)
-    trainer.fit(model, dm)
-    trainer.test()
+    with mlflow.start_run() as run:
+        trainer.fit(model, dm)
+        trainer.test()
```

Comment:
This change in the pytorch-lightning repo caused the issue:

https://github.com/PyTorchLightning/pytorch-lightning/pull/7232/files#diff-667a2513c158c2b01d73b479e4dea75587d8531f9cd286891d34570a2fd145dcR986-R987

Prior to this change, `Trainer.test` calls `Trainer.fit`. This means when `on_test_end` is called, an active run exists.
Now, `Trainer.test` no longer calls `Trainer.fit`. This means when `on_test_end` is called, NO active run exists and a new run will be created and leak.

```python
def on_test_end(self, trainer, pl_module):
    """
    Logs accuracy and other relevant metrics on the testing end

    :param trainer: pytorch lightning trainer instance
    :param pl_module: pytorch lightning base module
    """
    try_mlflow_log(mlflow.set_tag, "Mode", "testing")
    for key, value in trainer.callback_metrics.items():
        try_mlflow_log(mlflow.log_metric, key, float(value))
```